### PR TITLE
Use logging rather than syslog

### PR DESCRIPTION
**Pull request by jhart-r7**
This simply moves away from `syslog` so that this code will more readily work on non-Linux systems.  I tested this locally with one of our problematic bits of boto code (well, it isn't boto that is problematic so much as it is the S3-compatible service we are interfacing with...):

```
2018-03-14 11:08:31,052 root         WARNING    An error occurred (504) when calling the HeadObject operation (reached max retries: 4): Gateway Timeout: Retrying in 3 seconds...
```

I've also fixed a variety of other little things I noticed along the way that hinder development and testing.  Any feedback is much appreciated.

Thanks for looking!

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,10 +1,3 @@
 boto
 boto3
 botocore
-
-[dev]
-check-manifest
-
-[test]
```
Comment (linuxdynasty) [inline requirements.txt:8]: Why did you remove tests?

Comment (jhart-r7) [reply requirements.txt:8]: This syntax wasn't valid for me.  Per the instructions in README and general practice:

```
 $  pip install -r requirements.txt 
Invalid requirement: '[dev]'
Traceback (most recent call last):
  File "/Users/jhart/rapid7/awsretry/venv/lib/python2.7/site-packages/pip/req/req_install.py", line 82, in __init__
    req = Requirement(req)
  File "/Users/jhart/rapid7/awsretry/venv/lib/python2.7/site-packages/pip/_vendor/packaging/requirements.py", line 96, in __init__
    requirement_string[e.loc:e.loc + 8]))
InvalidRequirement: Invalid requirement, parse error at "u'[dev]'"
```

Comment (jhart-r7) [reply requirements.txt:8]: I believe I've fixed this.  Thanks for looking at this PR.  Really hoping to get this all fixed so we can more readily use this in some of our projects.
